### DISCUSSION TITLE
Batch size

### DISCUSSION THREAD

**Discussion by stijndebeer** [TUE-ARIA/NNCV#48] [Q&A] [answered]
So i was trying some implementation of my model and it is currently getting really big, so to train it properly I have to reduce the batch size to 4. Is there a way to increase the batch size while maintaining model complexity? My training is currently quite unstable compared to previous models although performance isn't too bad yet.

**Comment by ghost** [comment, marked as answer]
You could use automatic mixed precision (AMP) to reduce memory usage allowing for a larger actual batchsize on the gpu. If that's still not enough consider implementing gradient accumulation to increase your effective batchsize (accumulating gradients over multiple batches and updating weights after a specific number of batches).
